comskip.c

```diff
@@ -1960,11 +1960,34 @@ void InitHasLogo()
 
     int x,y;
     memset(haslogo, 0, MAXWIDTH*MAXHEIGHT*sizeof(char));
+    if (width > MAXWIDTH) 
+    {
+        fprintf(stderr, "Width exceeded max width %i >= %i\n", width, MAXWIDTH);
+        return;
+    }
     for (y = clogoMinY - LOGO_BORDER; y < clogoMaxY + LOGO_BORDER; y++)
     {
+	if (y >= MAXHEIGHT) 
+	{
+            fprintf(stderr, "Logo border exceeded max height %i >= %i\n", y, MAXHEIGHT);
+	    break;
+	}
         for (x = clogoMinX-LOGO_BORDER; x < clogoMaxX + LOGO_BORDER ; x++)

```

Comment:
Is the issue that x/y were becoming negative?

I think it would make sense to add the guards into the for loop expression instead of the body, something like:

```c
for (x = MIN(0, clogoMinX-LOGO_BORDER); x < MAX_WIDTH && x < clogoMaxX + LOGO_BORDER; x++)
```